File: 2610. Convert an Array Into a 2D Array With Conditions/solution.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<vector<int>> result;

        for (int num : nums) {
            bool found = false;

            for (auto& row : result) {
                if (find(row.begin(), row.end(), num) == row.end()) {
                    row.push_back(num);
                    found = true;
                    break;
                }
            }

            if (!found) {
                result.push_back({num});
            }
        }

        return result;
    }
};

File: 2610. Convert an Array Into a 2D Array With Conditions/solution.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function(nums) {
  const result = [];

      for (const num of nums) {
          let found = false;

          // Check if the current element is already in the result matrix
          for (const row of result) {
              if (!row.includes(num)) {
                  row.push(num);
                  found = true;
                  break;
              }
          }

          // If the element is not in any existing row, create a new row
          if (!found) {
              result.push([num]);
          }
      }

      return result;
};